### DISCUSSION TITLE
Manually write query to graphcache

### DISCUSSION THREAD

**Discussion by davidkhierl** [urql-graphql/urql#2445] [Q&A] [answered]
How to properly write query1 and map query2 mutation result to query1

my data looks like this
```js
currentUserQuery {
  currentUser: User
}

signInMutation {
  user: User
  accessToken: string
}
```

and on my `exhanges.cacheExchange.updates.mutation`
![image](https://user-images.githubusercontent.com/25083033/168321981-c8c8b041-b5bc-4a9f-85b9-b1f7a54ce104.png)

this work on signin but occasionally a warning is being thrown when triggering a refresh token
![image](https://user-images.githubusercontent.com/25083033/168322608-b96c6272-8d46-4386-9618-5a7f3925d643.png)

my refresh token is stored in cookies so no need to store on the state, here is my authExchange looks like
![image](https://user-images.githubusercontent.com/25083033/168323076-4eb84eac-4caa-4ab8-809c-d7698967cd3d.png)

I don't know if my implementation are correct I feel like I'm doing wrong here

**Comment by JoviDeCroock** [comment, marked as answer]
`result.signIn.user` is probably missing a `__typename` or if refreshToken is a different field then you probably need an updater too 😅 

i.e.

```
return { currentUser: { ...result.signIn.user, __typename: 'User'}  }
```

**Reply by davidkhierl** [reply]
Thank you for your response really appreciate it!

here is my refresh token mutation
![image](https://user-images.githubusercontent.com/25083033/168325563-9e49b666-3a56-4eda-b745-558baf57b1ca.png)

Do I need to explicitly include `__typename` property on all my mutation and queries?

**Reply by JoviDeCroock** [reply]
on the entities atleast, that's the essence for a normalized cache, documented here https://formidable.com/open-source/urql/docs/graphcache/normalized-caching/

**Reply by davidkhierl** [reply]
oh I see, I think I misunderstood this line here "Normalized caches can ask for the __typename field in selection sets automatically..." 😅

one last thing just need you opinion on my `updateQuery` if I'm doing it correctly.
![image](https://user-images.githubusercontent.com/25083033/168327620-be3c87b6-680d-4904-bea3-b7d5ad3f71c3.png)

do I need to explicitly include the `__typename` here?

Thank you so much!!!

**Reply by JoviDeCroock** [reply]
Normally you don't have to add `Query` no, some examples https://github.com/JoviDeCroock/threed-web/blob/master/src/client.js#L70

**Reply by davidkhierl** [reply]
very helpful example! again thank you!
